### PR TITLE
Add interpreter tests

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
By doing a similar change as `integration-tests.yml`, the pass rate actually decrease more, but the number of test cases increase:
`all: passed: 9062, skipped: 495, total: 10144, pass rate: 94.82%` from `all: passed: 8166, skipped: 409, total: 9162, pass rate: 95.23%`. The pass rate before the regression was `all: passed: 8481, skipped: 409, total: 9477, pass rate: 95.4%`.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -160,12 +160,12 @@ jobs:
           rm -rf ~/.triton
 
       - name: Run interpreter tests
-        env:
+        # env:
           # TRITON_INTERPRET: "1"
```
Comment (pbchekin) [inline .github/workflows/build_and_test.yml:164]: Why `TRITON_INTERPRET` is not set while in the upstream workflow it is set? Without setting we are running the subset (tests marked with `interpreter`) that already executed in the "Run core tests" step.

Comment (whitneywhtsang) [reply .github/workflows/build_and_test.yml:164]: There are extra failures in `test_flash_attention.py` that need further investigation.

Comment (pbchekin) [reply .github/workflows/build_and_test.yml:164]: May be we can temporary split interpreter tests and test_flash_attention and set `TRITON_INTERPRET` for the former?

Comment (whitneywhtsang) [reply .github/workflows/build_and_test.yml:164]: Changing to `TRITON_INTERPRET: "1"` is actually worse, 
```
$ TRITON_INTERPRET=1 python3 -m pytest -vvv -n 4 -m interpreter language/test_core.py --device xpu
=== 764 passed, 218 skipped, 82 warnings in 6.07s ===
```

Comment (pbchekin) [reply .github/workflows/build_and_test.yml:164]: Yeah, but we eventually need them in the workflow ...

Comment (whitneywhtsang) [reply .github/workflows/build_and_test.yml:164]: As long as we are ok to have degradation in pass rate.

Comment (pbchekin) [reply .github/workflows/build_and_test.yml:172]: We have to. The pass rate will go down again once we start counting modified tests (with `UserWarning FIXME`). IMHO we need to accept it.

Comment (whitneywhtsang) [reply .github/workflows/build_and_test.yml:172]: updated the test.

**Review by pbchekin — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
Latest result: `all: passed: 8936, skipped: 627, xfailed: 587, total: 10150, pass rate (w/o xfailed): 93.44%`

**Review by pbchekin — APPROVED** [review]

**Comment by etiotto** [comment]
Given that OpenAI runs "interpreter" test on device "cpu", I think to start we should do the same.

File: scripts/test-triton.sh [hunk 2]
```diff
@@ -116,13 +116,23 @@ run_core_tests() {
...
-  TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -n 8 --verbose --device xpu operators/
+  # run test_line_info.py separately with TRITON_DISABLE_LINE_INFO=0
+  TRITON_DISABLE_LINE_INFO=0 python3 -m pytest --verbose --device xpu language/test_line_info.py
   if [ $? -ne 0 ]; then
     echo "FAILED: return code $?" ; exit $?
   fi
 
-  # run test_line_info.py separately with TRITON_DISABLE_LINE_INFO=0
-  TRITON_DISABLE_LINE_INFO=0 python3 -m pytest --verbose --device xpu language/test_line_info.py
+  TRITON_INTERPRET=1 TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -vvv -n 4 -m interpreter language/test_core.py --device cpu
```
Comment (etiotto) [inline scripts/test-triton.sh:125]: @pbchekin when you parse the test results are we able to list the test results for the language tests twice, one for the case where the language tests are run with device "xpu" and the other with device "cpu". That way we can track the pass rate of these 2 categories.

Comment (whitneywhtsang) [reply scripts/test-triton.sh:125]: It is already happening, as we use different xml files for the two results.

**Pull request merged**